src/goto-programs/goto_program_template.h

```diff
@@ -516,99 +478,60 @@ void goto_program_templatet<codeT, guardT>::compute_loop_numbers()
 }
 
 template <class codeT, class guardT>
-void goto_program_templatet<codeT, guardT>::get_successors(
-  targett target,
-  targetst &successors)
+template <typename Target>
+std::list<Target> goto_program_templatet<codeT, guardT>::get_successors(
+  Target target) const
 {
-  successors.clear();
   if(target==instructions.end())
-    return;
+    return std::list<Target>();
 
-  targett next=target;
-  next++;
+  const auto next=std::next(target);
 
   const instructiont &i=*target;
 
   if(i.is_goto())
   {
-    for(const auto &t : i.targets)
-      successors.push_back(t);
+    std::list<Target> successors(i.targets.begin(), i.targets.end());
 
-    if(!i.guard.is_true() && next!=instructions.end())
+    if(i.guard.is_false() && next!=instructions.end())
```

Comment:
This is wrong - a `guard` is an expression an can be true, false, or something non-constant. Thus `!i.guard.is_true()` is *not* the same as `i.guard.is_false()`. That might well fix failing tests.